src/fuzz/protobufs/utils/spanner_emulator_ddl_statement_proto_to_string.cc

```diff
@@ -0,0 +1,188 @@
+        columnOptionsToString(column.allowcommittimestamp()));
+}
+
+// returns "data_type" or "ARRAY< data_type >" based on ColumnDataInfo
+std::string toString(ColumnDataInfo columnDataInfo) {
+    return columnDataInfo.isarray() ? 
+        absl::Substitute("ARRAY< $0 >", toString(columnDataInfo.scalartype(), 
+            columnDataInfo.length(), columnDataInfo.lengthtype())) :
+        toString(columnDataInfo.scalartype(), columnDataInfo.length(), 
+            columnDataInfo.lengthtype());
+}
+
+// returns the column's scalar type as a string
+std::string toString(ColumnDataInfo::ScalarType type, int length, 
+    ColumnDataInfo::LengthType lengthType) {
+    switch (type) {
+        case ColumnDataInfo::BOOL:
+            return "BOOL";
+        case ColumnDataInfo::INT64:
+            return "INT64";
+        case ColumnDataInfo::FLOAT64:
+            return "FLOAT64";
+        case ColumnDataInfo::STRING:
+            switch (lengthType) {
+                case ColumnDataInfo::BOUND:
+                    // ensures length is a positive number
+                    length = std::abs(length) + 1;
+                    return absl::Substitute("STRING( $0 )", 
+                        std::to_string(length % 2621440));
+                case ColumnDataInfo::UNBOUND:
+                    return absl::Substitute("STRING( $0 )", 
+                        std::to_string(length));
+                case ColumnDataInfo::MAX:
+                    return "STRING( MAX )";
+                default:
+                    return ""; // never triggers
+            }
+        case ColumnDataInfo::BYTES:
+            switch (lengthType) {
+                case ColumnDataInfo::BOUND:
+                    // ensures length is a positive number
+                    length = std::abs(length) + 1;
+                    return absl::Substitute("BYTES( $0 )", 
+                        std::to_string(length % 10485760));
+                case ColumnDataInfo::UNBOUND:
+                    return absl::Substitute("BYTES( $0 )", 
+                        std::to_string(length));
+                case ColumnDataInfo::MAX:
+                    return "BYTES( MAX )";
+                default:
+                    return ""; // never triggers
+            }
+        case ColumnDataInfo::DATE:
+            return "DATE";
+        case ColumnDataInfo::TIMESTAMP:
+            return "TIMESTAMP";
+        default: // never occurs given the protobuf structure
+            return "";
+    }
+}
+
+// if this column should be not null, return the appropriate string
+std::string isColumnNotNullToString(bool isNotNull) {
+    return isNotNull ? "NOT NULL" : "";
+}
+
+// return the options for this column as a string
+// the current api only allows for timestamps
+std::string columnOptionsToString(bool allowCommitTimestamps) {
+    std::string allowTimestamps = allowCommitTimestamps ? "true" : "null";
+    return absl::Substitute("{ OPTIONS ( allow_commit_timestamp = { $0 } ) }",
```

Comment:
I believe we don't want the brackets here.  IIRC, the syntax is something like:

"Column1 TIMESTAMP NOT NULL OPTIONS allow_commit_timestamp=false"

Also, we may actually want 3 states here--commit timestamps = true, commit timestamps = false, and commit timestamps not set.  However, for now it should be fine to just put a todo in.

Reply:
Leaving it as two options for now. Will implement later time-permitting otherwise will make a TODO for future